packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1791,7 +1791,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      */
     private async createRootDataStoreLegacy(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
         const fluidDataStore = await this._createDataStore(pkg, true /* isRoot */, rootDataStoreId);
-        fluidDataStore.bindToContext();
+        // back-compat 0.58.2000 - makeVisibleAndAttachGraph was added in this version to IFluidDataStoreChannel. For
+        // older versions, we still have to call bindToContext.
+        if (fluidDataStore.makeVisibleAndAttachGraph !== undefined) {
+            fluidDataStore.makeVisibleAndAttachGraph();
+        } else {
+            fluidDataStore.bindToContext();
+        }
```

Comment:
Why would this be undefined? fluidDataStore.makeVisibleAndAttachGraph !== undefined, if the container runtime is in 58.2000, then the datastores it creates would have to have makeVisibileAndAttachGraph as defined right? Is this back compat for older documents?

Reply:
FluidDataStoreRuntime (data store runtime layer) is in a different layer than ContainerRuntime and DataStoreContext (container runtime layer) and they can have different versions. We have to support N-2 compat across these 2 layers.